### ISSUE TITLE
"illegal instruction" Error when using gsl::span as Kernel Parameter

### ISSUE THREAD

**Issue by Spielix** [gsl-lite/gsl-lite#302]
This seems to happen since fb1236183dbbb1b54d0e6bc5c5b59fa974b8af14, so it's most probably due to changes to gsl_Expects?

I have observed it with CUDA 9.2 and gcc 6 from the Debian-Buster repositories.

Minimal example:
```cuda
#include <iostream>                                                                                                                                                                   [0/5952]
#include <gsl/gsl>

__global__ void mykernel(gsl::span<double> sp) {
        auto tid = blockDim.x * blockIdx.x + threadIdx.x;

        if (tid < sp.size()) 
                sp[tid] = static_cast<double>(tid);
}

int main() {
        int N = 512;
        double *d_ptr{};
        auto res = cudaMalloc(&ddata, N * sizeof(double));
        if (res != cudaSuccess) 
                std::cout << "cudaMalloc: " << cudaGetErrorString(res) << std::endl;

        mykernel<<<(N+255)/256, 256>>>(gsl::make_span(d_ptr, N));
          
        res = cudaDeviceSynchronize();
        if (res != cudaSuccess) 
                std::cout << "cudaDeviceSynchronize: " << cudaGetErrorString(res) << std::endl
;
        res = cudaGetLastError();
        if (res != cudaSuccess) 
                std::cout << "cudaGetLastError: " << cudaGetErrorString(res) << std::endl;

        cudaFree(d_ptr);
}
```
compiled with `nvcc -I/path/to/gsl-lite/include -o mininmal_example minimal_example.cu`

**Comment by mbeutel** [comment]
Thank you for the report. This was indeed a regression and should be fixed by https://github.com/gsl-lite/gsl-lite/commit/035f9d3c3a889f8681da3b52de049f669325a66e.

We have compile-only CI for CUDA, but unfortunately this bug surfaced only at runtime. Although I don't think any CI vendor offers CUDA hardware in an OSS plan, we should at least add some proper runtime tests to the test suite.

---

(The bug was not specific to any CUDA version, but since you mentioned CUDA 9.2 I'll point out that we currently don't test (and hence, do not support) anything older than CUDA 11.0. Is there a pressing need to support versions older than 11.0?

Also, please don't include the `<gsl/gsl>` header, which is for MS-GSL compatibility only; use `<gsl/gsl-lite.hpp>` instead.)

**Comment by Spielix** [comment]
We have a server for developement and small simulations which uses compute_35 architecture GPUs, so as far as I understand, we wont be able to update much further than CUDA 11, as their architectre is already deprecated in 11. That server was still on CUDA 7 or 8 a few weeks ago, so 9 is already good progress for us ;)  The main reason, that we don't have 11 yet is that we chose the easy way of using the package manager. Debian Buster has a backport of CUDA 11, but it doesn't seem complete (`compute sanitizer` didn't work for me when I tried it). I hope Debian Bullseye will get a release date soon.
On the cluster where I mainly work, CUDA 11 was added as a module just this month...
